internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -73,6 +77,33 @@ func (e *ELBV2) ListenerRuleHostHeaders(ruleARN string) ([]string, error) {
 	return hostHeaders, nil
 }
 
+// Rule wraps an elbv2.Rule to add some nice functionality to it.
+type Rule elbv2.Rule
+
+// DescribeRule returns the Rule with ruleARN.
+func (e *ELBV2) DescribeRule(ctx context.Context, ruleARN string) (Rule, error) {
+	resp, err := e.client.DescribeRulesWithContext(ctx, &elbv2.DescribeRulesInput{
+		RuleArns: aws.StringSlice([]string{ruleARN}),
+	})
+	if err != nil {
+		return Rule{}, err
+	} else if len(resp.Rules) == 0 {
+		return Rule{}, errors.New("not found")
```

Comment:
nit:
```suggestion
		return Rule{}, fmt.Errorf("rule ARN %s is not found", ruleARN)
```

Reply:
i think i'll leave it for now since the ruleARN context is added for both error cases [here](https://github.com/aws/copilot-cli/pull/4017/files#diff-924fba6d7ac2c8e2b099ec2776e78f958c7b225d5484001289b0e62283df3db1R248)